hlc/tests/test_planner.py

```diff
@@ -0,0 +1,145 @@
+sys.path.append('..')
+
+
+def test_empty_map_6x6():
+    keys = []
+    pos = (0, 0)
+
+    # create all keys aka all position in the grid
+    for i in range(6):
+        for j in range(6):
+            keys.append((i, j))
+
+    # create dic with all possible position, set value to False = pos not
+    # visited
+    all_positions = dict.fromkeys(keys, False)
+    all_positions[(0, 0)] = True
+    plan_output = planner.plan((6, 6), np.array([]), np.array([]), (0, 0))
+
+    # if position visited change value to True
+    for action in plan_output:
+        pos = planner.add_pos_tuple(pos, action)
+        all_positions[pos] = True
+
+    # for a map without obstacles the robot needs to visit all positions
+    for value in all_positions.values():
+        assert value == True
+
+
+def test_one_obstacle_6x6():
+    keys = []
+    pos = (0, 0)
+
+    # create all keys aka all position in the grid
+    for i in range(6):
+        for j in range(6):
+            keys.append((i, j))
+
+    # create dic with all possible position, set value to False = pos not
+    # visited
+    all_positions = dict.fromkeys(keys, False)
+    all_positions[(0, 0)] = True
+
+    plan_output = planner.plan(
+        (6, 6), np.array([[1, 2]]), np.array([]), (0, 0))
+
+    # if position visited change value to True
+    for action in plan_output:
+        pos = planner.add_pos_tuple(pos, action)
+        all_positions[pos] = True
+
+    # pos_blocked needs to be False, all other pos True --> robot explored the
+    # whole map with one obstacle
+    for key, value in all_positions.items():
+        if key == (
+                1, 2):  # pos with obsticle, value needs to be false = pos not visited
+            assert value != True
+        else:  # all other pos must be visited
+            assert value == True
+
+
+def test_multiple_obsticle_6x6():
+    keys = []
+    pos = (0, 0)
+
+    # create all keys aka all position in the grid
+    for i in range(6):
+        for j in range(6):
+            keys.append((i, j))
+
+    # create dic with all possible position, set value to False = pos not
+    # visited
+    all_positions = dict.fromkeys(keys, False)
+    all_positions[(0, 0)] = True
+
+    plan_output = planner.plan((6, 6), np.array(
+        [[0, 2], [1, 5], [3, 4], [5, 2]]), np.array([]), (0, 0))
+
+    # if position visited change value to True
+    for action in plan_output:
+        pos = planner.add_pos_tuple(pos, action)
+        all_positions[pos] = True
+
+    # pos_blocked needs to be False, all other pos True --> robot explored the
+    # whole map with multiple obstacles
+    for key, value in all_positions.items():
+        if key == (
+                0, 2):  # pos with obsticle, value needs to be False = pos not visited
+            assert value != True
+        elif key == (1, 5):
+            assert value != True
+        elif key == (3, 4):
+            assert value != True
+        elif key == (5, 2):
+            assert value != True
```

Comment:
Maybe it would make more sense to create a list containing all obstacles at the beginning e.g. `obstacles = [[0, 2], [1, 5], [3, 4], [5, 2]]` and to then to check here just if the key is in obstacles e.g.

```python
if key in obstacles:
    assert value != True
else:
    assert value == True
```